source/validate.cpp

```diff
@@ -326,7 +328,7 @@ spv_result_t ValidateBinaryUsingContextAndValidationState(
     spv_instruction_t inst;
     spvInstructionCopy(&binary->code[index], static_cast<SpvOp>(opcode),
                        wordCount, endian, &inst);
-    instructions.push_back(inst);
+    instructions.emplace_back(std::move(inst));
```

Comment:
but the copy (without std::move) needs an extra allocation which is then followed by a deallocation in the destructor of the original.